src/components/documentation/edit-tabs/edit-page-attached-resources.components.ts

```diff
@@ -0,0 +1,102 @@
+ * limitations under the License.
+ */
+
+import { IScope } from 'angular';
+import '@gravitee/ui-components/wc/gv-switch';
+import DocumentationService from '../../../services/documentation.service';
+import NotificationService from '../../../services/notification.service';
+
+interface IPageScope extends IScope {
+  fetcherJsonSchema: string;
+  rename: boolean;
+  editorReadonly: boolean;
+  currentTab: string;
+  currentTranslation: any;
+  acls: any;
+}
+
+const EditPageAttachedResourcesComponent: ng.IComponentOptions = {
+  bindings: {
+    gotoFunction: '=',
+    attachedResources: '<',
+    page: '<',
+  },
+  template: require('./edit-page-attached-resources.html'),
+  controller: function (
+    $mdDialog: angular.material.IDialogService,
+    DocumentationService: DocumentationService,
+    NotificationService: NotificationService,
+  ) {
+    'ngInject';
+
+    this.addAttachedResource = () => {
+      $mdDialog
+        .show({
+          controller: 'FileChooserDialogController',
+          controllerAs: 'ctrl',
+          template: require('../../dialog/fileChooser.dialog.html'),
+          clickOutsideToClose: true,
+          locals: {
+            title: 'Select a file to attach',
+            confirmButton: 'Add',
+          },
+        })
+        .then((response: any) => {
+          if (response.file) {
+            // upload new media to portal or api
+            const fd = new FormData();
+            let fileName = response.file.name;
+            if (response.filename) {
+              fileName = response.filename;
+            }
+            fd.append('file', response.file);
+            fd.append('fileName', fileName);
+
+            DocumentationService.addMedia(fd, this.page.id, this.apiId)
+              .then(() => this.gotoFunction())
+              .then(() => NotificationService.show(fileName + ' has been attached'));
+          }
+        });
+    };
+
+    this.removeAttachedResource = (resource: any) => {
+      $mdDialog
+        .show({
+          controller: 'DialogConfirmController',
+          controllerAs: 'ctrl',
+          template: require('../../dialog/confirmWarning.dialog.html'),
+          clickOutsideToClose: true,
+          locals: {
+            title: 'Would you like to remove "' + resource.fileName + '"?',
+            confirmButton: 'Remove',
+          },
+        })
+        .then((response) => {
+          if (response) {
+            this.page.attached_media = this.page.attached_media.filter(
+              (media) =>
+                !(media.mediaHash === resource.hash && media.mediaName === resource.fileName && media.attachedAt === resource.createAt),
+            );
+            DocumentationService.update(this.page, this.apiId)
+              .then(() => this.gotoFunction())
+              .then(() => NotificationService.show(resource.fileName + ' has been removed from page'));
+          }
+        });
+    };
+  },
+};
+
+export default EditPageAttachedResourcesComponent;
```

Comment:
We should avoid default export and do `export const EditPageAttachedResourcesComponent` direcly instead